migrations/educandu-2022-01-17-01-initial-schema.js

```diff
@@ -0,0 +1,196 @@
+    await createCollectionWithIndexes('documentLocks', [
+      {
+        name: '_idx_expires_',
+        key: { expires: 1 },
+        expireAfterSeconds: 0
+      }
+    ]);
+    await createCollectionWithIndexes('documentOrders', []);
+    await createCollectionWithIndexes('documentRevisions', [
+      {
+        name: '_idx_key_',
+        key: { key: 1 }
+      },
+      {
+        name: '_idx_order_',
+        key: { order: 1 },
+        unique: true
+      },
+      {
+        name: '_idx_key_order_',
+        key: { key: 1, order: 1 },
+        unique: true
+      }
+    ]);
+    await createCollectionWithIndexes('documents', [
+      {
+        name: '_idx_updatedOn_',
+        key: { updatedOn: -1 }
+      },
+      {
+        name: '_idx_slug_',
+        key: { slug: 1 }
+      },
+      {
+        name: '_idx_tags_',
+        key: { tags: 1 }
+      }
+    ]);
+    await createCollectionWithIndexes('lessons', [
+      {
+        name: '_idx_roomId_',
+        key: { roomId: 1 }
+      }
+    ]);
+    await createCollectionWithIndexes('maintenanceLocks', [
+      {
+        name: '_idx_expires_',
+        key: { expires: 1 },
+        expireAfterSeconds: 0
+      }
+    ]);
+    await createCollectionWithIndexes('migrations', []);
+    await createCollectionWithIndexes('passwordResetRequests', []);
+    await createCollectionWithIndexes('roomInvitations', [
+      {
+        name: '_idx_expires_',
+        key: { expires: 1 },
+        expireAfterSeconds: 0
+      },
+      {
+        name: '_idx_roomId_email_',
+        key: { email: 1, roomId: 1 },
+        unique: true
+      },
+      {
+        name: '_idx_token_',
+        key: { token: 1 },
+        unique: true
+      }
+    ]);
+    await createCollectionWithIndexes('roomLocks', [
+      {
+        name: '_idx_expires_',
+        key: { expires: 1 },
+        expireAfterSeconds: 0
+      }
+    ]);
+    await createCollectionWithIndexes('rooms', [
+      {
+        name: '_idx_owner_',
+        key: { owner: 1 }
+      },
+      {
+        name: '_idx_members_user_id_',
+        key: { 'members.userId': 1 }
+      }
+    ]);
+    await createCollectionWithIndexes('sessions', [
+      {
+        name: '_idx_expires_',
+        key: { expires: 1 },
+        expireAfterSeconds: 0
+      }
+    ]);
+    await createCollectionWithIndexes('settings', []);
+    await createCollectionWithIndexes('taskLocks', [
+      {
+        name: '_idx_expires_',
+        key: { expires: 1 },
+        expireAfterSeconds: 0
+      }
+    ]);
+    await createCollectionWithIndexes('tasks', [
+      {
+        name: '_idx_batch_id_',
+        key: { batchId: 1 }
+      },
+      {
+        name: '_idx_batch_id_processed_',
+        key: { batchId: 1, processed: 1 }
+      },
+      {
+        name: '_idx_task_id_processed_',
+        key: { taskId: 1, processed: 1 }
+      }
+    ]);
+    await createCollectionWithIndexes('users', []);
+    await this.db.collection('users').createIndexes([
```

Comment:
Why don't you use the helper function also to create the indexes here? :thinking:

Reply:
Guess! 😬

Reply:
:laughing: